pkg/connector/groups.go

```diff
@@ -93,6 +105,30 @@ func (o *groupBuilder) List(ctx context.Context, parentResourceID *v2.ResourceId
 	return outResources, nextPage, nil, nil
 }
 
+// parentIsSuperGroup validates that the parentResourceID received by the function call belongs to the corresponding super group.
+// The validation occurs between the ID on the group data (given by the API) and the ID of the parentResource.
+// superGroupId will be 0 (zero) if the group isn't a subgroup. parentResourceID will be nil if no parent resource was received in the List function.
+// Both cases are checked and handled in this function.
+func parentIsSuperGroup(superGroupId int, parentResourceID *v2.ResourceId) (bool, error) {
```

Comment:
It's not clear to me what a "super group" is based on reading gitlab documentation